### PR TITLE
Adds expression re-writing capabilities in LSP InitialFixup method.

### PR DESCRIPTION
**Pull request by jas-valgotar**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/IExpressionRewriter.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Expression re-writer interface. This will be leveraged by InitalFixup method to rewrite the expression in LSP.
+    /// </summary>
+    public interface IExpressionRewriter
+    {
+        public string Process(CheckResult check);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/IExpressionRewriter.cs:14]: >string [](http://example.com/codeflow?start=15&length=6)

can you comment this further - 
Assuming this returns an expression for rewritten result?

**Comment by MikeStall** [comment]
>         private readonly Dictionary<string, CodeFixHandler> _handlers = new Dictionary<string, CodeFixHandler>();

Can fixup be here (instead of engine)? 

EditorContextScope already has other state, like code fix handlers. 


---
Refers to: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/EditorContextScope.cs:42 in 30406d6. [](commit_id = 30406d6a7951c5d9b955098b09d5d3bea2d5d3a9, deletion_comment = False)

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Pull request merged**